SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -1155,14 +1155,45 @@ - (void)handleSendToMyself:(TSOutgoingMessage *)outgoingMessage
             [TSContactThread getOrCreateThreadWithContactId:contactId transaction:transaction];
         [cThread saveWithTransaction:transaction];
 
+        // We need to clone any attachments for message sent to self; otherwise deleting
+        // the incoming or outgoing copy of the message will break the other.
+        NSMutableArray<NSString *> *attachmentIds = [NSMutableArray new];
+        for (NSString *attachmentId in outgoingMessage.attachmentIds) {
+            TSAttachmentStream *_Nullable outgoingAttachment =
+                [TSAttachmentStream fetchObjectWithUniqueID:attachmentId];
+            OWSAssert(outgoingAttachment);
+            if (!outgoingAttachment) {
+                DDLogError(@"%@ Couldn't load outgoing attachment for message sent to self.", self.logTag);
+            } else {
+                TSAttachmentStream *incomingAttachment =
+                    [[TSAttachmentStream alloc] initWithContentType:outgoingAttachment.contentType
+                                                          byteCount:outgoingAttachment.byteCount
+                                                     sourceFilename:outgoingAttachment.sourceFilename];
+                NSError *error;
+                NSData *_Nullable data = [outgoingAttachment readDataFromFileWithError:&error];
```

Comment:
TIOLI: Especially for larger attachments, it would be nicer to copy on the filesystem rather than read and then write back the same data.

Reply:
I'm going to keep it simple here since this is a true edge case.